src/Types/Post.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace WPGraphQLPostsToPosts\Types;
+
+use WPGraphQLPostsToPosts\Interfaces\Hookable;
+use WPGraphQLPostsToPosts\Traits\ObjectsTrait;
+use WPGraphQLPostsToPosts\Types\Fields;
+
+class Post implements Hookable {
+
+	use ObjectsTrait;
+
+	public function register_hooks() : void {
+		add_action( get_graphql_register_action(), [ $this, 'register_where_input_fields' ] );
+		add_filter( 'graphql_map_input_fields_to_wp_query', [ $this, 'modify_query_input_fields' ], 10, 6 );
+	}
+
+	public function register_where_input_fields() : void {
+		$post_types = self::get_post_types();
+		foreach ( $post_types as $post_type ) {
+			$graphql_single_name = $post_type->graphql_single_name;
+
+			register_graphql_field(
+				'RootQueryTo' . $graphql_single_name . 'ConnectionWhereArgs',
+				Fields::NAME,
+				[
+					'type'        => [ 'list_of' => Fields::QUERY_TYPE ],
+					'description' => __( 'Id', 'wp-graphql-posts-to-posts' ),
+				]
+			);
+		}
+	}
+
+
+	public function modify_query_input_fields( array $query_args ) : array {
+		$p2p_connections_to_map = Fields::get_p2p_connections();
+
+		$field_names = [];
+		$post__in    = [];
+
+		$post_types = self::get_post_types();
+
+		foreach ( $post_types as $post_type ) {
+			$connection_name = $post_type->name;
+
+			$connections = array_filter( $p2p_connections_to_map, fn( $p2p_connection ) => $p2p_connection['from'] === $connection_name || $p2p_connection['to'] === $connection_name );
+
+			foreach ( $connections as $connection ) {
+				array_push( $field_names, $connection['name'] );
+			}
+		}
+
+		if ( ! isset( $query_args['postToPostConnections'] ) ) {
```

Comment:
Under what circumstances would `postToPostConnections` not be set? Returning early might not be the correct approach.